src/utils/__tests__/async-test.js

```diff
@@ -1,14 +1,47 @@
 /* @flow strict-local */
 import { sleep, tryUntilSuccessful } from '../async';
+import { Lolex } from '../../__tests__/aux/lolex';
 
 describe('sleep', () => {
+  let lolex: Lolex;
+
+  beforeAll(() => {
+    lolex = new Lolex();
+  });
+
+  afterAll(() => {
+    lolex.dispose();
+  });
+
+  afterEach(() => {
+    // clear any unset timers
+    lolex.clearAllTimers();
+  });
+
   test('waits for a given time in milliseconds', async () => {
     const expectedMs = 1000;
-    const start = Date.now();
-    await sleep(expectedMs);
-    const durationMs = Date.now() - start;
-    expect(expectedMs).toBeLessThanOrEqual(durationMs);
-    expect(durationMs).toBeLessThan(10 * expectedMs);
+
+    const expectationPromise = async () => {
+      const start = Date.now();
+      await sleep(expectedMs);
+      const durationMs = Date.now() - start;
+
+      expect(expectedMs).toBeLessThanOrEqual(durationMs);
+      // cannot be guaranteed: arbitrary delays are possible
+      // expect(durationMs).toBeLessThan(10 * expectedMs);
```

Comment:
This seems like a regrettable loss in test coverage: it means that if our `sleep` acquired a bug where, say, we were confused about units and always slept for as many seconds as we'd intended milliseconds, it'd still pass the test. Is there a reason that switching to Lolex means we need to sacrifice our test coverage of that failure mode?

It's true that a `sleep` in real life can't guarantee a timely wakeup. That's why the test only asks that the interval be within 10x of the target, which I think has been enough to make flakes nonexistent even on the very messy real systems that Travis CI runs our tests in. When we move to a spiffy mock setup where we have total control of the (virtual) environment, what I'd expect to happen is that the behavior only becomes more predictable, not less -- so that we could tighten this to like `1.01 * expectedMs`, and it would pass every time. That would be a very appealing kind of thing to be able to do when trying to test tricky timer-using code.

Reply:
That would be reasonable... so long as about half of this comment is in the code near the 10x test. Especially the bit where a 10x test is specifically intended to guard against a unit-scaling bug, rather than just being an unusually-large fudge factor.

(Done.)

Reply:
Well, I think the origin of it really is as a large fudge factor. :slightly_smiling_face: It had been 10%, and in 935c1f52f9795c2f516a9a1e8531ff0a50bf8143 we loosened it to 10x because that had flaked.

The possible unit-scaling bug is just a nice example of how some code like this plausibly could actually have a bug that this test with a large fudge factor would catch.